odilo.py

```diff
@@ -0,0 +1,598 @@
+        if 'notifiedHolds' in availability:
+            licenses_reserved = int(availability['notifiedHolds'])
+        else:
+            licenses_reserved = 0
+
+        # 'patrons_in_hold_queue' contains the number of patrons who are currently waiting for a copy of the book.
+        if 'holdsQueueSize' in availability:
+            patrons_in_hold_queue = int(availability['holdsQueueSize'])
+        else:
+            patrons_in_hold_queue = 0
+
+        return CirculationData(
+            data_source=DataSource.ODILO,
+            primary_identifier=primary_identifier,
+            licenses_owned=licenses_owned,
+            licenses_available=licenses_available,
+            licenses_reserved=licenses_reserved,
+            patrons_in_hold_queue=patrons_in_hold_queue,
+        )
+
+    @classmethod
+    def image_link_to_linkdata(cls, link, rel):
+        if not link:
+            return None
+
+        return LinkData(rel=rel, href=link, media_type=Representation.JPEG_MEDIA_TYPE)
+
+    @classmethod
+    def record_info_to_metadata(cls, book, availability):
+        """Turn Odilo's JSON representation of a book into a Metadata
+        object.
+
+        Note:  The json data passed into this method is from a different file/stream
+        from the json data that goes into the book_info_to_circulation() method.
+        """
+        if 'id' not in book:
+            return None
+
+        odilo_id = book['id']
+        primary_identifier = IdentifierData(Identifier.ODILO_ID, odilo_id)
+        active = book.get('active')
+
+        title = book.get('title')
+        subtitle = book.get('subtitle')
+        series = book.get('series')
+        series_position = book.get('seriesPosition')
+
+        contributors = []
+        author = book.get('author')
+        if author:
+            roles = [Contributor.AUTHOR_ROLE]
+            contributor = ContributorData(sort_name=author, display_name=author, roles=roles, biography=None)
+            contributors.append(contributor)
+
+        publisher = book.get('publisher')
+
+        # Metadata --> Marc21 260$c
+        published = book.get('publicationDate')
+        if not published:
+            # yyyyMMdd --> record creation date
+            published = book.get('releaseDate')
+
+        if published:
+            try:
+                published = datetime.datetime.strptime(published, "%Y%m%d")
+            except ValueError as e:
+                cls.log.warn('Cannot parse publication date from: ' + published + ', message: ' + e.message)
+
+        # yyyyMMdd --> record last modification date
+        last_update = book['modificationDate']
+        if last_update:
+            try:
+                last_update = datetime.datetime.strptime(last_update, "%Y%m%d")
+            except ValueError as e:
+                cls.log.warn('Cannot parse last update date from: ' + last_update + ', message: ' + e.message)
+
+        language = book.get('language')
```

Comment:
What format are your languages in? We should be able to handle it but I want to make sure.